happi/cli.py

```diff
@@ -252,6 +257,19 @@ def happi_cli(args):
 
         from IPython import start_ipython  # noqa
         start_ipython(argv=['--quick'], user_ns=devices)
+    elif args.cmd == "update":
+        # parse input
+        input_ = " ".join(args.json).strip()
+        if input_ == "-":
+            items_input = json.load(sys.stdin)
+        else:
+            items_input = json.loads(input_)
```

Comment:
Just to recap for `update` JSON variable args:
1. `-` to get input from stdin, a great feature
2. Assumes JSON string with optional quoting (as it effectively does something like`json.loads(' '.join(sys.argv[2:])`)

CLI API-wise I personally prefer requiring either pipe or filename (making supporting this really simple with a single https://docs.python.org/3/library/argparse.html#argparse.FileType - which supports files and stdin by way of `-`). 

Not a dealbreaker for merging this PR, and only reflects my opinion.

Reply:
@klauer I tried this, but I couldn't get it to work the way I want to

1) default behavior to be reading from stdin
2) able to use "$()" syntax

on the other hand you can always use files with the current implementation... `cat myfile.json | happi update` or `happi update "$(cat myfile.json)"`

Reply:
Right. My suggestion explicitly would not allow that additional input method, as I believe it should be done through a pipe (or a file).

@ZLLentz it's worth noting that if we are to move forward with this specific method, we will have to duplicate this functionality in all other tools we intend to support JSON with.